src/main/java/org/folio/service/inventory/InventoryInstanceManager.java

```diff
@@ -342,27 +346,19 @@ private Future<String> createInstance(JsonObject instanceRecJson, RequestContext
 
   public Future<CompositePoLine> openOrderHandleInstance(CompositePoLine compPOL, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
     return consortiumConfigurationService.getConsortiumConfiguration(requestContext)
-      .compose(configuration -> {
-        Future<String> instanceIdFuture;
-        if (compPOL.getInstanceId() != null) {
-          instanceIdFuture = Future.succeededFuture(compPOL.getInstanceId());
-        } else {
-          instanceIdFuture = getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext);
-        }
-        if (configuration.isEmpty()) {
-          return instanceIdFuture;
-        }
-        return instanceIdFuture
-          .compose(instanceId -> shareInstanceAmongTenantsIfNeeded(instanceId, configuration.get(), compPOL.getLocations(), requestContext));
-      }).map(compPOL::withInstanceId);
+      .compose(configuration ->
+        getInstanceRecord(compPOL, isInstanceMatchingDisabled, requestContext)
+          .compose(instanceId -> configuration
+            .map(consortiumConfiguration -> shareInstanceAmongTenantsIfNeeded(instanceId, consortiumConfiguration, compPOL.getLocations(), requestContext))
+            .orElse(Future.succeededFuture(instanceId))))
+      .map(compPOL::withInstanceId);
   }
 
   private Future<String> shareInstanceAmongTenantsIfNeeded(String instanceId, ConsortiumConfiguration consortiumConfiguration,
                                                            List<Location> locations, RequestContext requestContext) {
     return getTenantIdsWithoutShadowInstances(instanceId, consortiumConfiguration, locations, requestContext)
       .map(tenantIds -> tenantIds.stream()
-        .map(targetTenantId -> sharingInstanceService.createShadowInstance(instanceId, consortiumConfiguration,
-            RequestContextUtil.createContextWithNewTenantId(requestContext, targetTenantId)))
+        .map(tenantId -> createShadowInstanceIfNeeded(instanceId, consortiumConfiguration, createContextWithNewTenantId(requestContext, tenantId)))
```

Comment:
Directly calling `createShadowInstance` caused the issue, as it would not check for existing instances and attempt to create one directly. Call should be `createShadowInstanceIfNeeded` that does all the necessary checks and finally calls `createShadowInstance` if needed